src/App.tsx

```diff
@@ -1,61 +1,96 @@
+import { useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (userId: number): User | null => usersFromServer
+  .find(user => user.id === userId) ?? null;
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(
+    todosFromServer.map(todo => ({ ...todo, user: getUser(todo.userId) })),
+  );
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const isValid = /^[a-zA-Z0-9а-яА-Я\s]*$/.test(event.target.value);
```

Comment:
You use event.target.value several times. It's better to use restructuring in this case
const { value } = event.target;